airflow/providers/google/cloud/operators/vertex_ai/custom_job.py

```diff
@@ -180,6 +180,38 @@ def execute(self, context: Context) -> None:
             stacklevel=2,
         )
 
+    def execute_complete(self, context: Context, event: dict[str, Any]) -> dict[str, Any] | None:
+        if event["status"] == "error":
+            raise AirflowException(event["message"])
+        result = event["job"]
+        custom_job_id = self.hook.extract_custom_job_id_from_training_pipeline(result)
+        model_id = self.hook.extract_model_id_from_training_pipeline(result)
+        self.xcom_push(context, key="custom_job_id", value=custom_job_id)
+        if model_id:
+            self.xcom_push(context, key="model_id", value=model_id)
+            VertexAIModelLink.persist(context=context, task_instance=self, model_id=model_id)
+        else:
+            self.log.warning(
+                "It is impossible to get the Model ID. "
+                "The Training Pipeline did not produce a Managed Model because it was not "
+                "configured to upload a Model. Please ensure that the 'model_serving_container_image_uri' "
+                "and 'model_display_name' parameters are passed in when creating a Training Pipeline, "
+                "and check that your training script saves the model to os.environ['AIP_MODEL_DIR']."
+            )
+        return result
+
+    @cached_property
+    def hook(self) -> CustomJobHook:
+        return CustomJobHook(
+            gcp_conn_id=self.gcp_conn_id,
+            impersonation_chain=self.impersonation_chain,
+        )
+
+    def on_kill(self) -> None:
```

Comment:
@e-galan did you check that `on_kill` works fine for Custom* operators? I remember that we had a problem in the past when we moved this method to the base class.

Reply:
Will check. The rest of the methods certainly work fine.

Reply:
The method works for the sync mode and does not work for the deferrable mode (when the task execution is already in the triggerer), same as before.

Reply:
Will be addressed in a different PR, as it is a somewhat separate problem which affects many deferred operators. As of now there is no native way to make `on_kill()` work for the def mode. The existent solutions involve manipulation with asyncio inside the trigger class.